### PR TITLE
feat: Attach headers for SDK and App information to requests

### PR DESCRIPTION
**Pull request by corycaywood**
# Description
Add headers with app, device, and sdk information.

## Links
SDKCF-1237

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: remote-config/src/main/kotlin/com/rakuten/tech/mobile/remoteconfig/SdkHeadersInterceptor.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        appId = appId,
+        appName = context.packageName,
+        appVersion = context.packageManager
+            .getPackageInfo(context.packageName, 0)?.versionName ?: "NONE",
+        deviceModel = Build.MODEL,
+        deviceOsVersion = Build.VERSION.RELEASE,
+        sdkVersion = BuildConfig.VERSION_NAME
+    )
+
+    override fun intercept(chain: Interceptor.Chain): Response {
```
Comment (donnie-jp) [inline remote-config/src/main/kotlin/com/rakuten/tech/mobile/remoteconfig/SdkHeadersInterceptor.kt:31]: very nice!

File: remote-config/src/test/kotlin/com/rakuten/tech/mobile/remoteconfig/SdkHeadersInterceptorSpec.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                deviceOsVersion = "2.0.0",
+                sdkVersion = "3.0.0"
+            )
+        ).build()
+
+        client.newCall(
+            Request.Builder().url(server.url("")).build()
+        ).execute()
+
+        server.takeRequest().getHeader(name) shouldEqual value
```
Comment (donnie-jp) [inline remote-config/src/test/kotlin/com/rakuten/tech/mobile/remoteconfig/SdkHeadersInterceptorSpec.kt:58]: where does `value` get set?

Comment (corycaywood) [reply remote-config/src/test/kotlin/com/rakuten/tech/mobile/remoteconfig/SdkHeadersInterceptorSpec.kt:58]: Each array from the collection below gets expanded and passed as the values to the constructor.

**Review by donnie-jp — APPROVED** [review]
LGTM

**Pull request merged**